freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/UserKeytabService.java

```diff
@@ -48,6 +56,21 @@ private void validateSameAccount(String userAccountId, String environmentCrn) {
         }
     }
 
+    private void validateWorkloadUserInEnvironment(String workloadUsername, String environmentCrn) {
+        String accountId = Crn.safeFromString(environmentCrn).getAccountId();
+        FreeIpaClient freeIpaClient;
+        try {
+            freeIpaClient = freeIpaClientFactory.getFreeIpaClientByAccountAndEnvironment(environmentCrn, accountId);
+            Optional<User> user = freeIpaClient.userFind(workloadUsername);
+            if (user.isEmpty()) {
+                throw new BadRequestException(String.format("Could not find workload user %s within environment %s. Ensure that the user has " +
```

Comment:
Maybe just say: Actor %s has not been synced into environment %s"

Btw, what BadRequestException translates to the user?

Reply:
yeah maybe a notfoundex would be better

Reply:
BadRequestException is 400 bad request. I've updated the message (though I used workload user instead of actor her). And changed to NotFoundException.